### ISSUE TITLE
Additional row added on several schedules - lights coming on in the middle of the night

### ISSUE THREAD

**Issue by reidprichard** [nielsfaber/scheduler-card#671]
### Checklist

- [X] I checked for similar existing issues (both open and closed) before posting.
- [X] I will participate in further discussion about this issue and can help by testing (if requested).

### Card Version

v2.4.1

### Bug description

On two of six schedule schemes, an additional row in the scheme somehow showed up on its own. It does not seem possible to manually add this row, although it can be manipulated (and deleted) as usual.
![err1](https://user-images.githubusercontent.com/66481248/216505146-94e7df9d-f8b9-43ed-8acf-e0ce898968d2.png)
![err5](https://user-images.githubusercontent.com/66481248/216505169-6b1a988f-fa64-421b-b930-fb51fe56a449.png)

Unfortunately, the HA instance is not mine so I cannot say exactly when it showed up, but looking at the logs I believe I saw behavior indicating the bug at least as far back as 1/22.

The HA user switched out the bulb associated with one of the schedules recently, but the other schedule has remain untouched.

### Steps to reproduce

1. Install scheduler component
2. Set up several schedules
3. A month goes by
4. This issue is here

**Comment by nielsfaber** [comment]
Please check in developer tools -> states for the attributes of the affected schedule and post them here.
Note that if (unknown entity) is displayed, the entity is probably deleted and the schedule will not work anymore.

**Comment by reidprichard** [comment]
Here is the one with (unknown entity). The other was a bedroom lamp coming on in the middle of the night, so I had to fix it. However, I grabbed a backup before doing so, and I can restore the backup to a new VM if necessary.

I was told that this schedule was still working, which did not make sense to me either.

```
weekdays:
  - daily
timeslots:
  - 00:00:00 - 04:50:00
  - 04:50:00 - sunrise-00:00:00
  - sunrise-00:00:00 - sunset-00:00:00
  - sunset-00:00:00 - 20:00:00
  - 20:00:00 - 00:00:00
entities:
  - light.floor_lamp_in_dinette
actions:
  - service: light.turn_off
  - service: light.turn_on
    data:
      brightness: 102
  - service: light.turn_off
  - service: light.turn_on
  - service: light.turn_off
current_slot: 1
next_slot: 2
next_trigger: '2023-02-03T07:09:00-08:00'
tags: []
icon: mdi:calendar-clock
friendly_name: Dinette Floor Lamp AM/PM
```

**Comment by nielsfaber** [comment]
I have recreated the schedule locally:
![image](https://user-images.githubusercontent.com/16861605/216645902-1f9d6796-5438-4f03-b055-9f342ec58ad1.png)

```
weekdays: daily
timeslots:
- 00:00:00 - 04:50:00
- 04:50:00 - sunrise-00:00:00
- sunrise-00:00:00 - sunset-00:00:00
- sunset-00:00:00 - 20:00:00
- 20:00:00 - 00:00:00
entities: light.dining_lights
actions: 
- service: light.turn_off
- service: light.turn_on
- service: light.turn_off
- service: light.turn_on
- service: light.turn_off
current_slot: 2
next_slot: 3
next_trigger: 2023-02-03T17:30:00+01:00
tags: 
icon: mdi:calendar-clock
friendly_name: Schedule #5c8746
```

However, for me it shows up OK, regardless of how I resize the browser window.

Which means I am unable to reproduce the problem, and don't really know how to help at this point.
Obviously the sunrise / sunset times are different in my location, but as long these don't get past the 04:50:00 or 20:00:00 marks this shouldn't affect anything.

Did you try recreating the schedule to find some reproduction scenario for this issue? 
Also, did you try to view the card on a different browser / device?

**Comment by reidprichard** [comment]
Hmm, is there anything else I can do to debug?

The behavior is the same on Windows 11 (Firefox + Edge) and Android (Firefox).

What happens if sunrise/sunset get past those thresholds? I encountered another issue (I thought I opened an issue here but can't find it - maybe I'm just unfamiliar with GitHub) where sunrise/sunset are evaluated based on the local browser time rather than the instance time. For example, sunset is at 5:30pm in the instance's timezone, but I'm administering the instance from a timezone +3hrs. Therefore, when editing a schedule the card shows sunset as being at 8:30pm.
